pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1193,6 +1246,68 @@ func (nc *Controller) handleDisruption(zoneToNodeConditions map[string][]*v1.Nod
 	}
 }
 
+func (nc *Controller) podUpdated(oldPod, newPod *v1.Pod) {
+	if newPod == nil {
+		// Nothing to do when pod is deleted.
+		return
+	}
+	if len(newPod.Spec.NodeName) != 0 && (oldPod == nil || newPod.Spec.NodeName != oldPod.Spec.NodeName) {
+		nc.podUpdateQueue.Add(newPod)
+	}
+}
+
+func (nc *Controller) doPodProcessingWorker() {
+	for {
+		obj, shutdown := nc.podUpdateQueue.Get()
+		// "podUpdateQueue" will be shutdown when "stopCh" closed;
+		// we do not need to re-check "stopCh" again.
+		if shutdown {
+			return
+		}
+
+		pod := obj.(*v1.Pod)
+		nodeName := pod.Spec.NodeName
+
+		nc.processPod(nodeName, pod)
+	}
+}
+
+func (nc *Controller) processPod(nodeName string, pod *v1.Pod) {
+	defer nc.podUpdateQueue.Done(pod)
+
+	nodeHealth := nc.nodeHealthMap.getDeepCopy(nodeName)
+	if nodeHealth == nil {
+		// Pod will be handled by standard eviction pass,
+		// when node data is collected.
+		return
+	}
+
+	node, err := nc.nodeLister.Get(nodeName)
+	if err != nil {
+		klog.Errorf("Failed to read node %v.", nodeName)
+		return
+	}
+
+	_, currentReadyCondition := nodeutil.GetNodeCondition(nodeHealth.status, v1.NodeReady)
+	if currentReadyCondition == nil {
+		// We can simply return from here. Even if nodes cache is not up-to-date,
+		// the pod will be handled correctly (evicted if needed) during processing of the node update event.
+		return
+	}
+
+	pods := []*v1.Pod{pod}
+	// Taint based eviction is handled by TaintManager.
```

Comment:
This comment isn't extremely helpful - we do have processTaintBasedEviction and it is doing something.
Change to sth like:
// In taint-based eviction mode, only node updates are processed by NodeLifecycleController.
// Pods are processed by TaintManager

Reply:
Which actually made be realize, that we should only:
- register handlers for pods in taintBaseEviction mode
- start workers only in that mode

[Probably also rename some process* function to reflect that we only use them in NoTaintBasedEviction mode.

Reply:
Don't we need workers for marking pods not-ready? This part of code is executed independently of taintBaseEviction mode.

Reply:
Fair point - so just change the comment to what I added above.

Reply:
Done.